ftdc/ftdc.go

```diff
@@ -151,14 +153,47 @@ func (ftdc *FTDC) Remove(name string) {
 
 	for idx, statser := range ftdc.statsers {
 		if statser.name == name {
-			ftdc.logger.Debugw("Removed statser", "name", name, "type", fmt.Sprintf("%T", statser.statser))
+			ftdc.logger.Debugw("Removed statser", "name", name,
+				"type", fmt.Sprintf("%T", statser.statser), "generationId", ftdc.inputGenerationID)
 			ftdc.statsers = slices.Delete(ftdc.statsers, idx, idx+1)
 			ftdc.inputGenerationID++
 			return
 		}
 	}
 
-	ftdc.logger.Warnw("Did not find statser to remove", "name", name)
+	ftdc.logger.Warnw("Did not find statser to remove",
+		"name", name, "generationId", ftdc.inputGenerationID)
+}
+
+// conditionalRemoveStatser first checks the generation matches before removing the `name` Statser.
+func (ftdc *FTDC) conditionalRemoveStatser(name string, generationID int) {
+	ftdc.mu.Lock()
+	defer ftdc.mu.Unlock()
+
+	// This function gets called by the "write ftdc" actor. Which is concurrent to a user
+	// adding/removing `Statser`s. If the datum/name that created a problem came from a different
```

Comment:
Dumb question: when does a `Statser` get added while writing FTDC is occurring? I'm assuming reconfiguration of a robot such that an existing of new "subsystem" has a new `Stats` method? How could a reconfiguration cause that to happen?

Reply:
> when does a Statser get added while writing FTDC is occurring?

> I'm assuming reconfiguration of a robot such that an existing of new "subsystem" has a new Stats method?

Correct!

> How could a reconfiguration cause that to happen?

Not sure I follow, but maybe the statement your looking for is that I'm treating each component/service (i.e: resource manager resource) as a "subsystem". And the other PR I just put out maybe adds the clarity you're looking for here.

Reply:
I think I was curious whether an existing "subsystem" could have a new `Stats` implementation during the lifetime of a robot. I think the answer is yes, but only for modular resources.